frontend/src/modules/Shares/components/ShareBoxList.js

```diff
@@ -0,0 +1,489 @@
+        })
+      )
+      .then((response) => {
+        setRequestGroupOptions(
+          Array.from(
+            new Set(
+              response.data.getShareRequestsToMe.nodes.map(
+                (node) => node.principal.SamlGroupName
+              )
+            )
+          )
+        );
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch]);
+
+  const fetchInboxDatasetsOptions = useCallback(async () => {
+    setLoading(true);
+    const response = await client.query(
+      listDatasets(Defaults.selectListFilter)
+    );
+    if (!response.errors) {
+      setDatasets(
+        response.data.listDatasets.nodes.map((node) => ({
+          label: node.label,
+          value: node.datasetUri
+        }))
+      );
+    } else {
+      dispatch({ type: SET_ERROR, error: response.errors[0].message });
+    }
+    setLoading(false);
+  }, [client, dispatch]);
+
+  const fetchOutboxGroupAndDatasetOptions = useCallback(async () => {
+    await client
+      .query(
+        getShareRequestsFromMe({
+          filter: Defaults.selectListFilter
+        })
+      )
+      .then((response) => {
+        const alldatasets = response.data.getShareRequestsFromMe.nodes.map(
+          (node) => ({
+            label: node.dataset.datasetName,
+            value: node.dataset.datasetUri
+          })
+        );
+        setDatasets(
+          Array.from(
+            new Map(alldatasets.map((item) => [item['value'], item])).values()
+          )
+        );
+        setDatasetGroupOptions(
+          Array.from(
+            new Set(
+              response.data.getShareRequestsFromMe.nodes.map(
+                (node) => node.dataset.SamlAdminGroupName
+              )
+            )
+          )
+        );
+      })
+      .catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.Error });
+      })
+      .finally(() => setLoading(false));
+  }, [client, dispatch]);
+
+  useEffect(() => {
+    if (client) {
+      fetchMyGroupsOptions().catch((error) => {
+        dispatch({ type: SET_ERROR, error: error.message });
+      });
+      if (tab === 'inbox') {
+        if (!dataset) {
+          fetchInboxDatasetsOptions().catch((error) => {
+            dispatch({ type: SET_ERROR, error: error.message });
+          });
+        }
+        fetchInboxGroupOptions().catch((error) => {
+          dispatch({ type: SET_ERROR, error: error.message });
+        });
+      } else {
+        fetchOutboxGroupAndDatasetOptions().catch((error) => {
+          dispatch({ type: SET_ERROR, error: error.message });
+        });
+      }
+      setFilter(Defaults.filter);
+    }
+  }, [
+    client,
+    dispatch,
+    tab,
+    dataset,
+    fetchInboxDatasetsOptions,
+    fetchInboxGroupOptions,
+    fetchOutboxGroupAndDatasetOptions,
+    fetchMyGroupsOptions
+  ]);
+
+  useEffect(() => {
```

Comment:
Could this `useEffect()` call `fetchItems()` function - I think the logic is the exact same for both